OptimoveSDK/Sources/Classes/Optimobile/Optimobile.swift

```diff
@@ -116,6 +119,28 @@ class Optimobile {
         maybeAlignUserAssociation(initialUserId: initialUserId)
     }
 
+    static func completeDelayedConfiguration(config: OptimobileConfig) {
+        guard let credentials = config.credentials else {
+            Logger.warn("No credentials provided for delayed configuration completion.")
+            return
+        }
+        Logger.info("Completing delayed configuration with credentials: \(credentials)")
+        updateStorageValues(config)
+        setCredentials(credentials)
+        NotificationCenter.default.post(name: .optimobileInializationFinished, object: nil)
+    }
+
+    static func setCredentials(_ credentials: Credentials) {
+        Optimobile.instance?.credentials = credentials
+    }
+
+    static func updateStorageValues(_ config: OptimobileConfig) {
```

Comment:
do we need to update region? in android we dont because region shouldnt change / we already set it in the new builder init

Reply:
The region from init and credentials potentially can be different. I think we need to use the latest one to prevent unauthorized errors.

Reply:
my main assumption (reasonable?) was that region doesnt change, and if it does -- it's an error. 

Otherwise why do we even provide region to the builder? If region can change based on brand -- this means we wouldnt know it at start

Reply:
What if regions in a set of tenants' credentials differ?